### ISSUE TITLE
500s instead of 404s

### ISSUE THREAD

**Issue by tkazec** [cyberdelia/django-pipeline#439]
In our development environment, `pipeline` seems to return 500s when a file can't be found. This is an issue when using the Chrome dev tools, because it automatically tries to load (non-existent) .map files, which fills our logs up with tracebacks.

```
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.8_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/Users/teddy/.virtualenvs/buzzscreen/lib/python2.7/site-packages/django/contrib/staticfiles/handlers.py", line 65, in __call__
    return super(StaticFilesHandler, self).__call__(environ, start_response)
  File "/Users/teddy/.virtualenvs/buzzscreen/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 187, in __call__
    response = self.get_response(request)
  File "/Users/teddy/.virtualenvs/buzzscreen/lib/python2.7/site-packages/django/contrib/staticfiles/handlers.py", line 55, in get_response
    return self.serve(request)
  File "/Users/teddy/.virtualenvs/buzzscreen/lib/python2.7/site-packages/django/contrib/staticfiles/handlers.py", line 48, in serve
    return serve(request, self.file_path(request.path), insecure=True)
  File "/Users/teddy/.virtualenvs/buzzscreen/lib/python2.7/site-packages/django/contrib/staticfiles/views.py", line 38, in serve
    document_root, path = os.path.split(absolute_path)
  File "/Users/teddy/.virtualenvs/buzzscreen/bin/../lib/python2.7/posixpath.py", line 92, in split
    i = p.rfind('/') + 1
AttributeError: 'list' object has no attribute 'rfind'
[13/Mar/2015 01:31:12] "GET /static/ui/vendor/angular.min.js.map HTTP/1.1" 500 59
```

**Comment by mbrochh** [comment]
I'm having the same issue. Is there any recommended workaround?

**Comment by cyberdelia** [comment]
I don't see pipeline in the traceback, are you sure this is due to pipeline?

**Comment by skolsuper** [comment]
+1 also having this issue. These are my staticfiles finders:

```
STATICFILES_FINDERS = (
        'pipeline.finders.FileSystemFinder',
        'django.contrib.staticfiles.finders.AppDirectoriesFinder',
        'pipeline.finders.CachedFileFinder',
        'pipeline.finders.PipelineFinder',
    )
```

I tried commenting them out one at a time and the 500 error disappeared when `CachedFileFinder` was commented out (although that left me with 404s instead). I will investigate further when I have time, for now I'm still stuck on pipeline1.3 :(

**Comment by skolsuper** [comment]
Okay so I had time now after all.  I think it's something to do with [this line](https://github.com/cyberdelia/django-pipeline/blob/master/pipeline/finders.py#L47)
```
start, _, extn = path.rsplit('.', 2)
```
Which, as far as I can tell, is supposed to separate out the cache-busting bit of the filename and throw it away.  It's wrapped in a `try: except ValueError` (maybe ought to change this to just wrap one line) which presumably is supposed to fail for paths that aren't cached files.  However I have named my output files `main.min.css` and `app.min.js` so it won't fail there.  I don't know why passing non-existent paths to `find` would cause this error though.

Maybe it would be better to use a regex to separate out the cache-busting bit anyway. I am not familiar with this stuff to make any suggestions really. Hope this info helps identify the issue.

**Comment by skolsuper** [comment]
This bug was actually fixed in PR #379. with added tests. I have submitted another PR (#481) that brings it up to date with master, as the github account seems dormant now.

**Comment by skolsuper** [comment]
I've not much experience with this sort of thing yet, but if you need someone to QA some pull requests @cyberdelia I would love to help if I can.
